radixdlt/src/main/java/com/radixdlt/middleware2/TokenFeeLedgerAtomChecker.java

```diff
@@ -165,6 +165,26 @@ private boolean isUpAndForFee(SpunParticle sp) {
 		return false;
 	}
 
+	// Check that there is at most one output particle and no input particle is smaller
+	private boolean noSuperfluousParticles(ImmutableList<Pair<Spin, TransferrableTokensParticle>> transferableTokens) {
+		final Map<Spin, List<TransferrableTokensParticle>> particlesBySpin = transferableTokens.stream()
+				.collect(Collectors.groupingBy(Pair::getFirst, Collectors.mapping(Pair::getSecond, Collectors.toList())));
+
+		final List<TransferrableTokensParticle> inputParticles = particlesBySpin.getOrDefault(Spin.DOWN, List.of());
+		final List<TransferrableTokensParticle> outputParticles = particlesBySpin.getOrDefault(Spin.UP, List.of());
+
+		if (outputParticles.isEmpty()) {
+			return true;
+		} else {
+			final UInt256 outputAmount = outputParticles.get(0).getAmount();
```

Comment:
Code from `else` block can be moved out of block, so nesting will be reduced.

Reply:
heh, guess I need to adjust back to more imperative style of code ;) to me this:
```
if (abc) return cde
else return xyz;
```
looks much prettier than
```
if (abc) return cde;

return xyz;
```

Reply:
It's not about functional vs imperative but rather "early cut off" approach. The idea is to make check which may prevent code from execution as soon as possible and leave function early, preserving "happy day scenario" at the top nesting level and therefore make it easier to read and follow.